crates/bevy_image/src/image.rs

```diff
@@ -1504,14 +1554,180 @@ impl Image {
+            });
+        };
+
+        Self::load_color(bytes, self.texture_descriptor.format)
+    }
+
+    #[inline]
+    fn set_color_at_internal(
+        &mut self,
+        coords: UVec3,
+        color: Color,
+    ) -> Result<(), TextureAccessError> {
+        let format = self.texture_descriptor.format;
+
+        let Some(bytes) = self.pixel_bytes_mut(coords) else {
+            return Err(TextureAccessError::OutOfBounds {
+                x: coords.x,
+                y: coords.y,
+                z: coords.z,
+            });
+        };
+
+        Self::store_color(bytes, format, color)
+    }
+}
+
+/// Iterator returned by [`Image::rect_bytes`].
+#[derive(Clone)]
+pub struct RectBytes<'a> {
+    data: slice::ChunksExact<'a, u8>,
+    range: Range<usize>,
+    format: TextureFormat,
+}
+impl RectBytes<'_> {
+    /// Gets the texture format for the rect.
+    pub fn format(&self) -> TextureFormat {
+        self.format
+    }
+}
+impl<'a> Iterator for RectBytes<'a> {
+    type Item = &'a [u8];
+    fn next(&mut self) -> Option<&'a [u8]> {
+        self.data.next().map(|s| &s[self.range.clone()])
+    }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (self.len(), Some(self.len()))
+    }
+}
+impl<'a> DoubleEndedIterator for RectBytes<'a> {
+    fn next_back(&mut self) -> Option<&'a [u8]> {
+        self.data.next_back().map(|s| &s[self.range.clone()])
+    }
+}
+impl<'a> ExactSizeIterator for RectBytes<'a> {
+    fn len(&self) -> usize {
+        self.data.len()
+    }
+}
+impl<'a> FusedIterator for RectBytes<'a> {}
+
+/// Iterator returned by [`Image::rect_bytes_mut`].
+pub struct RectBytesMut<'a> {
+    data: slice::ChunksExactMut<'a, u8>,
+    range: Range<usize>,
+    format: TextureFormat,
+}
+impl RectBytesMut<'_> {
+    /// Gets the texture format for the rect.
+    pub fn format(&self) -> TextureFormat {
+        self.format
+    }
+
+    /// Directly copies the bytes from another rect to this one.
+    ///
+    /// Will fail if the rects are different sizes or formats. Completely
+    /// ignores transparency and will directly overwrite the bytes.
+    pub fn copy_from(self, other: RectBytes<'_>) -> Result<(), TextureCopyError> {
+        let dest = UVec2::new(self.range.len() as u32, self.data.len() as u32);
+        let src = UVec2::new(other.range.len() as u32, other.data.len() as u32);
+        if dest != src {
+            return Err(TextureCopyError::SizeMismatch { dest, src });
+        }
+
+        if self.format != other.format {
+            return Err(TextureCopyError::FormatMismatch {
+                src: self.format,
+                dest: other.format,
+            });
+        }
+
+        for (dest, src) in self.zip(other.clone()) {
+            dest.copy_from_slice(src);
+        }
+        Ok(())
+    }
+
+    /// Copies the pixels from another rect to this one by performing color conversion.
+    ///
+    /// Will fail if the rects are different sizes or formats. Completely
```

Comment:
This does not seem to be correct, conversion/translation does happen with different formats.

Reply:
Yup, that was a typo; I copied the docs from the other funciton incorrectly.